### PR TITLE
CB-17273 added unit test to verify BlackListed version checking logic…

### PR DESCRIPTION
**Pull request by lnardai**
Added version normalizer, so it will convert 7.2.15-1.cdh7.2.15.p1.26792553 to 7.2.15 which can be understood by the BlackListedDownScaleRole and BlackListedUpScaleRole

See detailed description in the commit message.

### CODE REVIEW

**Comment by lnardai** [comment]
retest this please!

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java [hunk 1]
```diff
@@ -353,6 +360,44 @@ public void testValidationIfNodeManagerCountWillBeHigherThanZeroInTheSameGroup()
                 Optional.of(clouderaManagerRepo), hostGroup, -2, Set.of(compute, worker));
     }
 
+    @Test
+    public void checkKafkaBrokerBlackListedUpscaleShouldBeAllowed() {
```
Comment (bergerdenes) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java:364]: I would simplify it with a parameterized test, also if we have a cutover on the version for a feature, I would test 1 or 2 versions less, the cutover version, and 1 or 2 versions more.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java [hunk 2]
```diff
@@ -353,6 +360,44 @@ public void testValidationIfNodeManagerCountWillBeHigherThanZeroInTheSameGroup()
...
+        componentsByHostGroup.put("compute", components);
+        when(processor.getComponentsByHostGroup()).thenReturn(componentsByHostGroup);
+        Optional<ClouderaManagerProduct> product = Optional.of(clouderaManagerRepo);
+        Throwable exception = Assert.assertThrows(BadRequestException.class,
+                () -> subject.validateBlackListedScalingRoles(ACCOUNT_ID, processor, hostGroup, 1, product));
+        assertEquals("'KAFKA_BROKER' service is not enabled to scale until CDP 7.2.12", exception.getMessage());
+    }
+
+    @Test
+    public void isVersionEnablesScaling() {
```
Comment (bergerdenes) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java:396]: I would simplify it with a parameterized test, also if we have a cutover on the version for a feature, I would test 1 or 2 versions less, the cutover version, and 1 or 2 versions more.

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java (hunk 1 above)
Comment (lnardai) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java:364]: I've provided a test for both cases allowed/disallowed. This logic has only 2 possible outcomes so I feel like the parametrised test is an overkill and makes the test less readable. 
Regardless this is wolf's domain, so I'll make the changes.

**Comment by lnardai** [comment]
Sorry, had some problems with rebasing.

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**